Arf.77862/vary ARP.CLIENT_IDS by environment

gabezurita opened the pull request:
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary
Allow ARP client-ids to vary by environment and fix ARP sign-out.

Reference: https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/prod/values.yaml#L2367

Note: in staging, sign-in works, but sign-out is currently broken because of the following:
https://github.com/department-of-veterans-affairs/vets-website/blob/2c165632e0d3d430ec0e48b5dd51da64d61d6ea8/src/applications/accredited-representative-portal/constants/index.js#L11-L15

### Steps to reproduce
1. On https://staging.va.gov/representative, login as `vets.gov.user+1@gmail.com`
2. Click on the user nav on the header and then click `Sign Out`
3. The browser sign out URL that gets constructed is `https://staging-api.va.gov/v0/sign_in/logout?client_id=arp`
4. `client_id=arp` in staging results in the error `{"errors":"Client id is not valid"}`

The error gets generated in [this line](https://github.com/department-of-veterans-affairs/vets-api/blob/5eaf8343593c04af3ea305ae66676372934b41ff/app/controllers/v0/sign_in_controller.rb#L177-L179) when a [SignIn::ClientConfig.find_by(client_id: `arp`) is not found here](https://github.com/department-of-veterans-affairs/vets-api/blob/5eaf8343593c04af3ea305ae66676372934b41ff/app/controllers/v0/sign_in_controller.rb#L356). Passing in the correct client_id via the URL should resolve the issue.

The changes in [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/30878) or the [alternative fix PR](https://github.com/department-of-veterans-affairs/vets-website/pull/30901/files) should fix that error. I'm open to whatever solution Identity thinks would work better. Please let me know if you want an in-person explanation of the above or if you want to pair so we can investigate further!

See screenshots:
![image](https://github.com/user-attachments/assets/64b53082-66d2-4b0b-b7e1-d31c6e6e8f7c)
![image](https://github.com/user-attachments/assets/30e10f53-6b3b-4084-9b4f-47f09faccac9)
![image](https://github.com/user-attachments/assets/1b7e777a-3c31-4f81-af91-7c1ba450416c)

## Related issue(s)
https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/77862

Slack Canvas describing how it was figured out what needed to be updated to get this working in a staging environment:
https://dsva.slack.com/docs/T03FECE8V/F07CXK78K08

Ultimately, the below change was the only needed one:
https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/2968/files

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Acceptance criteria

### Quality Assurance & Testing

- [X] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [X] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [X] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

gabezurita committed:
add ARP prod client_id

raudabaugh reviewed (commented):
Thanks for this! While you're at it, should we add the clientId to the dev environment config? I see the "Client id is not valid" message there too.

gabezurita commented:
> Thanks for this! While you're at it, should we add the clientId to the dev environment config? I see the "Client id is not valid" message there too.

Hi Sam! Thanks for testing this. The SignIn::ClientConfig for the dev environment comes from seeds and not [the associated vsp-infra-application-manifests yaml file](
https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/dev/values.yaml#L2053-L2064). Can you try running the development seeds file in your local and see if that resolves the issue? See:
https://github.com/department-of-veterans-affairs/vets-api/blob/5095b955f6423b23b0a47400a4790d66a9719e16/db/seeds/development.rb#L128-L140

Alternatively, we could add an `arp_client_id` value to the vsp-infra-application-manifests dev config, as was done in [vsp-infra-application-manifests/apps/vets-api/staging/values.yml](https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/staging/values.yaml#L2362). That change would happen below:
https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/dev/values.yaml#L2053-L2064

cc @rileyanderson, should we make that addition?

rileyanderson commented:
> > Thanks for this! While you're at it, should we add the clientId to the dev environment config? I see the "Client id is not valid" message there too.
> 
> ~Hi Sam! Thanks for testing this. The SignIn::ClientConfig for the dev environment comes from seeds and not [the associated vsp-infra-application-manifests yaml file](https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/dev/values.yaml#L2053-L2064). Can you try running the vets-api development seeds file in your local and see if that resolves the issue? See: https://github.com/department-of-veterans-affairs/vets-api/blob/5095b955f6423b23b0a47400a4790d66a9719e16/db/seeds/development.rb#L128-L140~
> 
> Sam meant https://dev.va.gov/representative! We'll need to add an `arp_client_id` value to the vsp-infra-application-manifests dev config, as was done in [vsp-infra-application-manifests/apps/vets-api/staging/values.yml](https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/staging/values.yaml#L2362). That change would happen below: https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/dev/values.yaml#L2053-L2064
> 
> @rileyanderson, can we make that addition?

We can just use the default `arp` on `dev` since it's mocked and similar to localhost. It is enabled now

gabezurita committed:
refactor per-environment CLIENT_IDS handling

gabezurita commented on src/platform/user/authentication/config/prod.config.js:
```diff
@@ -85,4 +86,27 @@ export default {
+    },
+    oAuthOptions: {
+      ...arpWebOAuthOptions,
+      clientId: 'fe0d4b2cac7935e7eec5946b8ee31643', // SOURCE: https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/prod/values.yaml#L2367
```
I added a note and will make a ZH issue!
https://github.com/department-of-veterans-affairs/vets-website/blob/62e1a58f945567c35ff896bc40d3ba25c4bfd726/src/platform/utilities/oauth/constants.js#L36-L38

gabezurita replied:
Here's the ZH issue!
https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/zh/475

gabezurita committed:
Merge branch 'main' into arf.77862/add-prod-client-id

* main:
  replace fa references (#30882)
  Bump vets-json-schema version (#30871)
  replace fa with va-icon (#30864)
  [SIMPLE_FORMS] fix: list-and-loop - misc display bug fixes (#30478)
  [CST] [Frontend] Added required prop to 5103 document upload form (#30887)
  VACMS-18418: Update search app links to va-link component (#30856)
  Form flow and review page updates (#30865)
  remove arial label (#30880)
  update forms field in Direct deposit (#30827)
  Update Caregivers application to utilize updated schema (#30773)
  updated display of save link on error msg (#30870)
  added radio to topic and subtopic is less than 15 items (#30866)
  add unit tests (#30860)
  Claimant Contact Phone Email screen (#30787)

gabezurita committed:
Merge branch 'main' into arf.77862/add-prod-client-id

* main:
  Simplified MHV Landing Page for unregistered and unverified users (#30881)

ianmagenta reviewed (changes requested):
Despite the code comments on `oAuthOptions` in `staging.config.js`, @asg5704's comments on [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/28225#discussion_r1509574674) seem to suggest that we want to keep the hashes for the ARP `clientId` in the config files themselves. 

My recommendation would be to leave `staging.config.js` and `oauth/constants.js` as they are and set up the changes needed in `prod.config.js` (adding the prod `clientId` under `oAuthOptions`).

gabezurita commented:
> Despite the code comments on `oAuthOptions` in `staging.config.js`, @asg5704's comments on [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/28225#discussion_r1509574674) seem to suggest that we want to keep the hashes for the ARP `clientId` in the config files themselves.
> 
> My recommendation would be to leave `staging.config.js` and `oauth/constants.js` as they are and set up the changes needed in `prod.config.js` (adding the prod `clientId` under `oAuthOptions`).

@asg5704 and @ianmagenta, I can do that! 😄 Out of curiosity, mind if I ask the benefit (or reasoning) behind using the environments files (`dev/staging/prod.config.js`) instead of a conditional within `oauth/constants.js`?

gabezurita commented:
@department-of-veterans-affairs/octo-identity, I’m having a CI failure on some identity vets-website code that’s not failing locally. The [unit test that fails in CI](https://github.com/department-of-veterans-affairs/vets-website/actions/runs/9980901782/job/27583544270?pr=30878) (but not locally) is platform/user/tests/authentication/components/DowntimeBanner.unit.spec.js, though the failure occurs in the test wrapper platform/utilities/tests/api/index.unit.spec.js. I see that identity encountered the [same problem in this Identity Slack thread](https://dsva.slack.com/archives/CSFV4QTKN/p1718203751597339). I’ve added log statements and do not see any differences between the CI environment and my local. What are the steps to troubleshoot/fix that failing unit test?

See failure below:
```
running:  src/platform/user/tests/authentication/components/DowntimeBanner.unit.spec.js
  1) test wrapper
       apiRequest
         should not return JSON on (status: 204):
     TypeError: Cannot read property 'response' of undefined
      at _callee4$ (src/platform/utilities/tests/api/index.unit.spec.js:63:23)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (src/platform/utilities/tests/api/index.unit.spec.js:37:103)
      at _next (src/platform/utilities/tests/api/index.unit.spec.js:39:194)

  2) test wrapper
       apiRequest
         should not return JSON on (status: 404):
     TypeError: Cannot read property 'response' of undefined
      at /home/runner/work/vets-website/vets-website/src/platform/utilities/tests/api/index.unit.spec.js:79:25
      at processTicksAndRejections (internal/process/task_queues.js:93:5)

  3) test wrapper
       apiRequest
         should return JSON on (status: 403):
     TypeError: Cannot read property 'response' of undefined
      at /home/runner/work/vets-website/vets-website/src/platform/utilities/tests/api/index.unit.spec.js:97:25
      at processTicksAndRejections (internal/process/task_queues.js:93:5)

  4) test wrapper
       apiRequest
         should not fail when downloading a file:
     TypeError: Cannot read property 'response' of undefined
      at _callee7$ (src/platform/utilities/tests/api/index.unit.spec.js:141:23)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (src/platform/utilities/tests/api/index.unit.spec.js:37:103)
      at _next (src/platform/utilities/tests/api/index.unit.spec.js:39:194)
```

ddzz commented on src/platform/utilities/oauth/constants.js:
```diff
@@ -25,15 +25,30 @@ export const API_SIGN_IN_SERVICE_URL = ({
+    // https://github.com/department-of-veterans-affairs/vets-website/pull/30878#discussion_r1681519493
+    if (environment.isProduction()) return 'fe0d4b2cac7935e7eec5946b8ee31643'; // SOURCE: https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/prod/values.yaml#L2367
+    if (environment.isStaging()) return 'ce6db4d7974daf061dccdd21ba9add14'; // SOURCE: https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/staging/values.yaml#L2362
+    if (environment.isDev()) return 'arp'; // NOTE: Same as localhost: https://github.com/department-of-veterans-affairs/vets-website/pull/30878#issuecomment-2233940395
+    return 'arp';
```
Seems like these lines could be simplified to just `return 'arp'`?

gabezurita replied:
Yeah, I was thinking it would be good to explicitly call out `if (environment.isDev()) return 'arp';` so it's clear that `isDev` was considered in this implementation given confusion around that environment (see [comment 1](https://github.com/department-of-veterans-affairs/vets-website/pull/30878#issuecomment-2233876507) and [comment 2](https://github.com/department-of-veterans-affairs/vets-website/pull/30878#issuecomment-2233940395)). I could just remove `if (environment.isDev()) return 'arp';` and make a clarifying comment.

ddzz commented:
@gabezurita Most likely you're not seeing the unit tests fail locally because you're running them against a dev build. There's a process to run them against the production build locally, but I haven't done it since 2022, so I'd have to ask platform support how to do that.

ianmagenta commented:
> @asg5704 and @ianmagenta, I can do that! 😄 Out of curiosity, mind if I ask the benefit (or reasoning) behind using the environments files (`dev/staging/prod.config.js`) instead of a conditional within `oauth/constants.js`?

Since the overall goal is to configure things for specific environments, I think it makes the most sense that the changes themselves be in the config files rather than the constants.

> NOTE: if we go the route you are suggesting, we'll also need to update the `SIGNT_OUT_URL` generating code to be environment-aware, as it references `SIS.CLIENT_IDS.ARP` from `oauth/constants.js`:
> 
> https://github.com/department-of-veterans-affairs/vets-website/blob/2c165632e0d3d430ec0e48b5dd51da64d61d6ea8/src/applications/accredited-representative-portal/constants/index.js#L11-L15

It's not 100% clear to me why this change would need to be made (since I'm guessing this works with the `clientId` in the staging config). I'm assuming we just want to add a different `clientId` value for production and (if that's the case) then everything should work the same (once the values are added to the prod config). If there's something beyond that (or something I'm missing) please feel free to let me know.

> I thought the more DRY, central place to change would be in the [oauth/constants.js#CLIENT_IDS](https://github.com/department-of-veterans-affairs/vets-website/blob/1ba7d153d41a0b0731369aca534d2b63acbdae90/src/platform/utilities/oauth/constants.js#L28-L35) const instead of having the conditional in [src/applications/accredited-representative-portal/constants/index.js#SIGN_OUT_URL](https://github.com/department-of-veterans-affairs/vets-website/blob/01a834b9c046aad113667b08de8506519ec160b8/src/applications/accredited-representative-portal/constants/index.js#L23-L27).
> 
> For clarity, if we go with the above-suggested route, this is what the implementation would look like — do we prefer the below solution (happy to go with whatever Identify prefers)? https://github.com/department-of-veterans-affairs/vets-website/pull/30901/files

Thanks for posting the draft PR! I think the comments I left above apply to the draft as a whole. Again, feel free to let me know if I'm not understanding things correctly.

gabezurita commented:
@ianmagenta, in regards to the below:
> It's not 100% clear to me why this change would need to be made (since I'm guessing this works with the clientId in the staging config). I'm assuming we just want to add a different clientId value for production and (if that's the case) then everything should work the same (once the values are added to the prod config). If there's something beyond that (or something I'm missing) please feel free to let me know.

Logging in works, but logging out is currently broken in staging because of the following:
https://github.com/department-of-veterans-affairs/vets-website/blob/2c165632e0d3d430ec0e48b5dd51da64d61d6ea8/src/applications/accredited-representative-portal/constants/index.js#L11-L15

Steps to reproduce:
1. On https://staging.va.gov/representative, login as `vets.gov.user+1@gmail.com`
2. Click on the user nav on the header and then click `Sign Out`
3. The browser sign out URL that gets constructed is `https://staging-api.va.gov/v0/sign_in/logout?client_id=arp`
4. `client_id=arp` in staging results in the error `{"errors":"Client id is not valid"}`

The changes in [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/30878) or the [alternative fix PR](https://github.com/department-of-veterans-affairs/vets-website/pull/30901/files) should fix that error. Please let me know if you want to pair so we can investigate further!

See screenshots:
![image](https://github.com/user-attachments/assets/64b53082-66d2-4b0b-b7e1-d31c6e6e8f7c)
![image](https://github.com/user-attachments/assets/30e10f53-6b3b-4084-9b4f-47f09faccac9)
![image](https://github.com/user-attachments/assets/1b7e777a-3c31-4f81-af91-7c1ba450416c)

joeniquette reviewed (changes requested):
There is a lot to confirm within what appears to be a small change. I have a few folks out on PTO this week so I'd like to request you hold off on merging any changes here until at least Monday when my lead frontend engineer can review this.

ianmagenta commented:
@gabezurita - Thanks for clarifying things! I didn't notice that detail in the PR summary or ticket, so it may be worth adding that info. My thought would be to go with the [draft PR](https://github.com/department-of-veterans-affairs/vets-website/pull/30901) (pending some revisions). But, like Joe mentioned, I'll wait for our lead frontend engineer to weigh in.

gabezurita commented:
> @gabezurita - Thanks for clarifying things! I didn't notice that detail in the PR summary or ticket, so it may be worth adding that info. My thought would be to go with the [draft PR](https://github.com/department-of-veterans-affairs/vets-website/pull/30901) (pending some revisions). But, like Joe mentioned, I'll wait for our lead frontend engineer to weigh in.

Thanks for the feedback! I added that to the PR description and await the Identity lead frontend engineer's feedback.

gabezurita committed:
Merge branch 'main' into arf.77862/add-prod-client-id

* main: (81 commits)
  added error checking to session url (#31000)
  Adding feature toggle to Find a Rep (#30985)
  88434-Remove Unused Maintance Code in Dashboard.jsx (#30904)
  Replace actions/create-release in daily deploy workflow (#30978)
  Update remaining va-process-list web components to the USWDS v3 version (#30948)
  Remove unnecessary aria label on sign-in button to Form Upload (#30970)
  Removes profile_contacts toggle from featureFlagNames.json (#30949)
  added new tests for code coverage (#30984)
  update last updated date and add tests (#30945)
  82397 - VAOS timezone unit test fix (#30971)
  #VEBT-315 #Comments added alert for verifing account if user is not yet verified (#30991)
  Ask VA 1213 - Category and topic summary box issue (#30955)
  update focus by adding ref to alert, update contact option to lowercase) (#30958)
  Updated medical supplies link in MHV Landing Page(#30980)
  Add clinic phone number with extension to details page (#30969)
  #87738: Create Pending appts page with unscheduled referral and expired referral (#30968)
  VACMS 18486 - PAW title and test (#30944)
  Changes instances of usa-label to use  vads-u-background-color--primary (#30959)
  fix issue with null destination address (#30960)
  MHV-59972/adds-endpoint-for-krames (#30842)
  ...

asg5704 commented:
@gabezurita here are my recommendations:

- [ ] Revert the `oauth/constants.js` changes, they are not needed
- [ ] Add the specific `clientId` in the `oAuthOptions` for both staging & production (as @ianmagenta suggested)
- [ ] Update your `SIGN_OUT_URL` function to look at the clientId stored in session storage `sessionStorage.getItem('ci')`

asg5704 commented:
> @gabezurita here are my recommendations:
> 
> * [ ]  Revert the `oauth/constants.js` changes, they are not needed
> * [ ]  Add the specific `clientId` in the `oAuthOptions` for both staging & production (as @ianmagenta suggested)
> * [ ]  Update your `SIGN_OUT_URL` function to look at the clientId stored in session storage with `sessionStorage.getItem('ci')`

@gabezurita @department-of-veterans-affairs/accredited-representation-management

gabezurita commented:
> > @gabezurita here are my recommendations:
> > 
> > * [ ]   Revert the `oauth/constants.js` changes, they are not needed
> > * [ ]   Add the specific `clientId` in the `oAuthOptions` for both staging & production (as @ianmagenta suggested)
> > * [ ]   Update your `SIGN_OUT_URL` function to look at the clientId stored in session storage with `sessionStorage.getItem('ci')`
> 
> @gabezurita @department-of-veterans-affairs/accredited-representation-management

@asg5704, thanks for looking into this. I just returned from PTO and am excited to implement the suggested route tomorrow!

gabezurita committed:
implement per env oauth clientIDs Identity Team feedback
See: https://github.com/department-of-veterans-affairs/vets-website/pull/30878#issuecomment-2256505980
- Revert the oauth/constants.js changes, they are not needed
- Add the specific clientId in the oAuthOptions for both staging & production (as @ianmagenta suggested)
- Update your SIGN_OUT_URL function to look at the clientId stored in session storage with sessionStorage.getItem('ci')

gabezurita commented:
> @gabezurita here are my recommendations:
> 
> * [x]  Revert the `oauth/constants.js` changes, they are not needed
> * [x]  Add the specific `clientId` in the `oAuthOptions` for both staging & production (as @ianmagenta suggested)
> * [x]  Update your `SIGN_OUT_URL` function to look at the clientId stored in session storage with `sessionStorage.getItem('ci')`

Happy Friday, @asg5704! I made the suggested Identity Team changes in [this commit](https://github.com/department-of-veterans-affairs/vets-website/pull/30878/commits/5e02c0873398083cd2c9b8272c637bac155e43bc). How does [the overall change](https://github.com/department-of-veterans-affairs/vets-website/pull/30878/files) look now?

asg5704 reviewed (dismissed):
LGTM @gabezurita thank you

gabezurita committed:
Merge branch 'main' into arf.77862/add-prod-client-id

* main: (191 commits)
  Add duplicate of SC for testing (#31192)
  MHV-48219 MHV-48206 Implement front end data gap fixes Pathology (#30786)
  Remove custom font styles for landing page alert and intro text (#31104)
  MBMS-66418 initial commit and removal of prod flag (#30994)
  Vebt 75 del date (#31008)
  remove reference to font-awesome icon (#31201)
  component-library@v44.0.0 (#31200)
  Add functionality for redirect on stale session (#31183)
  Sponsor place of brirth no longer required (#31167)
  remove font-awsome fonts (#30629)
  makes VAMC use the GET endpoint vs the POST (#31114)
  Update test coverage for Caregivers app helper methods (#31195)
  Update test coverage for Caregivers app form config methods (#31193)
  88082 Add the provider/practitioner name under the Who section in the details page redesign (#31176)
  MBMS-65005 Self-Service: Pre-Need Integration: Military History Section: Military Ranks align with Service Start & End Date (#30973)
  VEBT-327  Feature toggle to show the correct button for 1995 or 5495. (#31084)
  Check in/82910/med content (#31138)
  MHV-60060: PHR refresh fixes (#31187)
  Update test coverage for the Caregivers app submit transformer (#31175)
  DR | Increase shared code coverage (#31177)
  ...

gabezurita committed:
Merge branch 'main' into arf.77862/add-prod-client-id

* main: (46 commits)
  adds setUpServer functionality to downtimeBanner test (#30378)
  Update HLR subtitle (#31256)
  Assets Review Error Handling (#31162)
  MHV-60852 Secure Messaging Send button Datadog tag defect
  89908a Removed unused marriage schema logic - form 10-10d (#31190)
  Add upload documents page (#31204)
  10-7959f-1 Remove Prefill (#31171)
  Removes logic that was breaking the InfoAlert display (#31243)
  Ask VA/Mock auth exp and custom pages (#31232)
  MHV-60213: Labs and tests and care summaries and notes mocks set up (#31224)
  SC testing | Update app, start & intro (#31237)
  remove fa icon no longer used on site (#31238)
  Updates Personal health care contacts instructions to call HEC (#31165)
  Remove status improvement canceled flag (#31228)
  Move sign in modal query params to Redux action (#31146)
  VACMS-18798 Homepage redesign modal: remove custom analytics (#31207)
  remove reference to font awesome icon (#31236)
  Added deduction code description to debt detail page (#31209)
  VACMS 18800 - remove recordEvent and related variables (#31210)
  remove old reference to font awesome (#31235)
  ...

gabezurita committed:
Merge branch 'main' into arf.77862/add-prod-client-id

* main:
  Discharge Upgrade Wizard v2 cleanup (#31244)
  New SCO page changes and cypress test (#31265)
  gracefully fail when content-build registry is not (#31286)
  MHV 60368 pilot single message (#31284)
  Form 10-10EZR keyboard only test (#31079)
  reverts to working DowntimeBanner.unit.spec.js + refactors (#31285)
  Check in/89875/add start end date to request (#31275)
  Remove sign in from Form Upload widget (#31254)
  VACMS-16936 DUW v2 Updates to add optional step (#31215)
  set progress bar to first in focus (#31261)
  trying to get rid of prefill (#31271)
  Fix VAOS Appointment details unit test (#31253)
  Implement VerifyAndRegisterAlert and MhvBasicAccountAlert to MHV Landing page - version 2 (#31029)
  Mhv 59281 contact list connect api (#31252)
  Updated TERA content for special characters (#31234)
  Simple Forms - re-work the intro page in form upload flow (#31259)
  made some updates to bring test coverage up to date (#31245)
  Update the alert message for request flow when the services is down (#31249)
  Ask VA 1245 - Update health care label to constant (#31246)

gabezurita committed:
Merge branch 'main' into arf.77862/add-prod-client-id

* main:
  Vebt 378 disable 22 5495 (#31283)
  Character of Discharge: Refactor and require option select (#31250)

gabezurita commented:
Update: This PR is blocked due to a CI issue unrelated to it. Please see this [#vsp-identity thread for details](https://dsva.slack.com/archives/CSFV4QTKN/p1723138268912429).

I will merge this PR as soon as that issue is resolved. cc @department-of-veterans-affairs/octo-identity and @department-of-veterans-affairs/benefits-accredited-rep-facing-engineers

gabezurita committed:
Merge branch 'main' into arf.77862/add-prod-client-id

* main: (67 commits)
  UL only for multi pre-check-in (#31363)
  Benefits Questionnaire - Mobile Styling (#31334)
  Add green task form to auth patterns form app (#31338)
  90102 Removing check_in_experience_45_minute_reminder (#31350)
  89801 On desktop, set detail page content width to 8 columns (#31359)
  Refactors MHV Landing Page alerts (#31280)
  89386 Update Page Titles (Telehealth Appointment Video and Phone - Part 3) (#31304)
  MHV-61012: Labs and tests mocks fixes (#31360)
  Import royalties and other properties chapter (#31257)
  89951 Update content and layout of missing documents overview page - form 10-7959c (#31345)
  Update PR template app removal steps (#31292)
  MBMS-64991 added styling to the 0247-pmc.scss which targets this exact header in those two pages (#31269)
  VACMS 12070 - wait times widget h3 -> h4 (#31349)
  Mhv 60734 spacing changes about page (#31337)
  MHV-61012: Medical records user bypass mock implemented (#31342)
  MHV-60719-investigate-med-info-pdf-txt-print (#31242)
  Appoint a representative contact screens (#31226)
  VFEP-1671 and VFEP-1672 add links for 'How to Use GI Bill Benefits' and 'VetSuccess on Campus (VSOC)' (#31327)
  Add introduction page link url (#31341)
  update @department-of-veterans-affairs/generator-vets-website to 3.9.2 (#31343)
  ...

gabezurita committed:
revert all but arp/constants/index.js changes to get CI green

gabezurita committed:
Merge branch 'main' into arf.77862/add-prod-client-id

gabezurita commented:
## Update

After much troubleshooting CI (see [this Slack thread for details](https://dsva.slack.com/archives/CSFV4QTKN/p1723138268912429)), we discovered that the `src/applications/accredited-representative-portal/constants/index.js` [change passes CI if done alone](https://github.com/department-of-veterans-affairs/vets-website/compare/arf.77862/add-prod-client-id-test-ci?expand=1), but not when in conjunction with changes to `src/platform/user/authentication/config/staging.config.js` and `src/platform/utilities/oauth/constants.js` (even if the changes to said files are [mere `// comment` changes](https://github.com/department-of-veterans-affairs/vets-website/compare/arf.77862/add-prod-client-id-test-ci-5?expand=1)). Thus, @asg5704 and I have agreed to split the changes in this PR to multiple branches to see if subsequent changes will pass CI.

It's possible that merging the change to `src/applications/accredited-representative-portal/constants/index.js` will cause subsequent changes to `src/platform/user/authentication/config/staging.config.js` and `src/platform/utilities/oauth/constants.js` to go red. If that is the case, we will revert.
 
Given the green CI, after [reducing the amount of files updated](https://github.com/department-of-veterans-affairs/vets-website/pull/30878/commits/79f84f39529f3a74c5fcebf52972a3d31895c698) in this PR and [merging `main` again (there was another flaky test)](https://github.com/department-of-veterans-affairs/vets-website/actions/runs/10405496100/job/28816912688?pr=30878), I'm re-requesting approval from @asg5704 and @ddzz!

cc @raudabaugh

ddzz commented on src/platform/user/authentication/config/prod.config.js:
```diff
@@ -85,4 +86,27 @@ export default {
+    },
+    oAuthOptions: {
+      ...arpWebOAuthOptions,
+      clientId: 'fe0d4b2cac7935e7eec5946b8ee31643',
```
This ID is for staging only, correct? If so, will we need a new ID for production?

ddzz replied:
Also, would it make sense to move this value to a `const`?

gabezurita replied:
This is the `prod` clientId. See:
https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/prod/values.yaml#L2367

gabezurita replied:
> Also, would it make sense to move this value to a const?
The current configuration was decided after discussing it with Identity. For context on this, see:
* https://github.com/department-of-veterans-affairs/vets-website/pull/30878#pullrequestreview-2187045731
* https://github.com/department-of-veterans-affairs/vets-website/pull/30878#issuecomment-2239716572
* https://github.com/department-of-veterans-affairs/vets-website/pull/30878#issuecomment-2256505980

ddzz replied:
My idea would be to move this to `src/platform/utilities/oauth/constants.js`, but if this is the agreed-upon approach with Identity then that's fine with me.
